doc/hardware/peripherals/rtc.rst

```diff
@@ -6,11 +6,78 @@ Real-Time Clock (RTC)
 Overview
 ********
 
-This is a placeholder for API specific to real-time clocks.  Currently
-all RTC peripherals are implemented through :ref:`counter_api` with
-device-specific API for counters with real-time support.
+A real-time clock tracks the date and time using an internal calendar.
```

Comment:
I've seen two types of RTC HW - with internal calendar an without (so they only provides precise battery-backed counter in seconds or milliseconds).

It worth to mention in documentation that currently zephyr supports only first type of RTC and currently zephyr doesn't provide calendar helpers for software calendar implementation for second RTC type.

Reply:
The RTC API is specifically for real-time clocks. The other type of hardware is a real-time counter, which just happens to share the same initials, hence this clarification in the docs :)

A real-time counter is a fancy name for a low power counter, which is clearly distinguished from a real-time clock as you can't set the value of a counter, just, in some cases, clear it. They don't necessarily run at 1Hz either, see https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.nrf52832.ps.v1.1%2Frtc.html

I am unsure what benefit there is to mention, in the documentation for the real-time clock API, that some real-time clock drivers have previously been forced to use the counter API. If you could provide me with some exact wording on what you would like me to add or alter, that would be helpful :) Maybe add a line saying "This should not be confused with a real-time counter, which uses the counter API <reference to counter API>?"

Reply:
> The other type of hardware is a real-time counter, which just happens to share the same initials, hence this clarification in the docs :)

I'm not sure if such separation exist. At least I know that it's not strictly followed :)
As example (the first thing which came to mind): STMicroelectronics STM32F103xx, STM32F102xx, STM32F101xx, STM32F105xx, STM32F107xx series.

The peripheral is named `Real-time clock (RTC)` but it doesn't have internal calendar - so basically it's battery-backed counter with alarm functionality. [Proof](https://www.st.com/resource/en/reference_manual/rm0008-stm32f101xx-stm32f102xx-stm32f103xx-stm32f105xx-and-stm32f107xx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf), see chapter 18 on 482 page.

> I am unsure what benefit there is to mention

It would be beneficial for new developers who planning to add their HW support. And in general is would be useful to specify explicitly what we do / don't support with current API given the vague separation between real-time counter / real-time clock :)

Reply:
In that case, I think we should do the following:
* Expand the doc to both describe how this API defines an RTC, and specifically point out that it does not support real-time counters, while explaining the distinction between them.
* Expand the doc to explain that this API is negating the need to use the counter API for RTCs as it has been done in the past.

Do you have more points to add more before I update it?

Reply:
Some devices are both.  My RV3032 driver has support for both calendar alarms and countdown style alarms.

Not supporting counter type alarms means the driver needs to pretend to be two devices so another API can be added.  There needs to be a dummy dts node for the other device.  Method to start/top the counter have issues, since call on one device affects the other too.

Reply:
Documentation has been updated to clearly define and clarify what is meant by RTC in this API. A history of RTCs in Zephyr has also been added to the docs.